src/Middleware/Session/src/DistributedSession.cs

```diff
@@ -105,7 +106,7 @@ private byte[] IdBytes
                     _sessionIdBytes = new byte[IdByteCount];
                     RandomNumberGenerator.Fill(_sessionIdBytes);
                 }
-                return _sessionIdBytes;
+                return _sessionIdBytes!;
```

Comment:
:\ Not a huge fan of this suppression, but I'm not exactly sure what the alternative would be. You _could_ put `MemberNotNullWhen` on the `IsAvailable` property and mark this property as nullable, but I'm not sure it's an improvement over what you have here. Here are all the potential options I see:

* Applying `MemberNotNullWhen` to `IsAvailable`
* Changing the property type to `byte[]?`
* Returning `_sessionIdBytes ?? Array.Empty<byte>()`
* Returning `_sessionIdBytes ?? throw new InvalidOperationException(...)`
* Keeping the suppression

Reply:
@Tratcher thoughts?

Reply:
This code could be restructured. Load always sets _sessionIdBytes or _sessionId so if IdBytes is called then it will return a non-null value. How about just removing the IsAvailable check here?